### PR TITLE
Journal utils merger

### PR DESCRIPTION
**Pull request by daisieh**
Merged DryadSubmissionJournalUtils into JournalUtils.

### CODE REVIEW

**Comment by daisieh** [comment]
I added some fixes to the journal-submit parsing, just so we get thrown more useful exceptions than NullPointerExceptions. This has nothing to do with any other fix, but I thought I'd just lump it in here anyway.

**Comment by daisieh** [comment]
I fixed some exception-handling in DryadEmailSubmission and a tiny bug in EmailParserForManuscriptCentral.java. This has nothing to do with any other fix, but I thought I'd just lump it in here anyway.

I tested this by adding the `vm-journal-submit` label to the emails with the label `problem-test-messages` in the dryad.journal.submit gmail account, then running https://localhost:9999/journal-submit/retrieve. You should get something like the following:

```
2015-09-15 03:33:06,592 INFO  org.datadryad.submission.DryadEmailSubmission @ retrieving mail with label 'vm-journal-submit'
2015-09-15 03:33:07,921 INFO  org.datadryad.submission.DryadEmailSubmission @ retrieved 2 messages
2015-09-15 03:33:07,926 INFO  org.datadryad.submission.DryadEmailSubmission @ MIME contentType/ID/encoding: text/plain; charset=UTF-8 null quoted-printable
2015-09-15 03:33:08,897 INFO  org.datadryad.submission.DryadEmailSubmission @ Exception thrown while processing MIME message: No submission ID found in message, org.datadryad.submission.SubmissionException
2015-09-15 03:33:08,897 INFO  org.datadryad.submission.DryadEmailSubmission @ MIME contentType/ID/encoding: text/plain; charset=UTF-8; format=flowed null null
2015-09-15 03:33:08,909 INFO  org.datadryad.submission.DryadEmailSubmission @ Exception thrown while processing MIME message: Concept not found for journal ecog', org.datadryad.submission.SubmissionException
```

**Comment by daisieh** [comment]
The travis failure is the usual null fail of a DSpace unit test.

File: dspace/modules/journal-submit/src/main/java/org/datadryad/submission/DryadEmailSubmission.java [hunk 1]
```diff
@@ -343,136 +342,132 @@ private String parseToXML (ParsingResult result) {
...
+            result.setJournalCode(journalCode);
+            result.setJournalName(journalName);
+            // Do this because this is what the parsers are expecting to
+            // build the corresponding author field from
+            for (Address address : addresses) {
+                result.setSenderEmailAddress(address.toString());
             }
 
-            LOGGER.debug("Registered journal: " + journal.toString());
+            if (context != null) {
```
Comment (ryscher) [inline dspace/modules/journal-submit/src/main/java/org/datadryad/submission/DryadEmailSubmission.java]: This needs to be in a `finally` clause.

**Comment by ryscher** [comment]
Code looks good. I didn't test running this, but I'm ok to merge it after the context close is moved into `finally`.

**Comment by daisieh** [comment]
@ryscher: apparently that is one of the fixes that was on my working branch but not my merge branch! It's fixed now.

**Pull request merged**